sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2659,6 +2660,19 @@ abstract class SQLQuerySuiteBase extends QueryTest with SQLTestUtils with TestHi
       }
     }
   }
+
+  test("SPARK-36180: Support TimestampNTZ type in Hive") {
+    withTable("tb") {
+      val dt = "2018-11-17 13:33:33.0"
+      val ddl =
+        s"CREATE TABLE tb as SELECT TIMESTAMP_LTZ'$dt' as c0, TIMESTAMP_NTZ '$dt' as c1"
```

Comment:
I think the only problem is we can't create hive-compatible table with timestamp ntz. This seems like an acceptable limitation. What do you think? @sarutak @MaxGekk

Reply:
Hive serde table is very special, as it includes both data source and metastore. For metastore, the current decision is to store the schema as it is in a hive-incompatible way. For the data source part, this PR proposes to store NTZ as Hive's timestamp.

I think there are 2 options
1. do not support NTZ at all for hive serde tables
2. revisit the decision for metastore. We can still store table with ntz in a hive-compatible way, by storing both ntz and ltz as hive timestamp

Reply:
Yeah, it's a Hive's limitation. So, Spark can't support `timestamp_ntz` for Hive provider.
If user need to store a table which contains `timestamp_ntz` as a Hive table, the user should manually convert `timestamp_ntz` to `timestamp` if it's really necessary.

Reply:
I am +1 with option 2.  The benefit is that users can create/read tables with TimestampNTZ columns.